Neos.Flow/Classes/Property/TypeConverter/PersistentObjectConverter.php

```diff
@@ -215,6 +222,9 @@ public function convertFrom($source, $targetType, array $convertedChildPropertie
                 );
                 throw new InvalidTargetException($exceptionMessage, 1297935345);
             }
+            if ($isCollectionType && $originalValue !== ObjectAccess::getProperty($object, $propertyName)) {
+                throw new TypeConverterException(sprintf('You overwrote the collection property "%s" of type "%s". This will lead to data loss. Read more about it and how to prevent this here: ', $propertyName, $targetType), 1602324196);
```

Comment:
TODO before merge:
```suggestion
                throw new TypeConverterException(sprintf('You overwrote the collection property "%s" of type "%s". This will lead to data loss. Read more about it and how to prevent this here: ADDLINKHERE', $propertyName, $targetType), 1602324196);
```